qt/scientific_interfaces/Indirect/IndirectFunctionBrowser/ConvFunctionModel.cpp

```diff
@@ -0,0 +1,689 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ConvFunctionModel.h"
+#include "MantidAPI/FunctionFactory.h"
+#include "MantidAPI/IFunction.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/MultiDomainFunction.h"
+#include "MantidQtWidgets/Common/FunctionBrowser/FunctionBrowserUtils.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace IDA {
+
+using namespace MantidWidgets;
+using namespace Mantid::API;
+
+namespace {}
+
+ConvFunctionModel::ConvFunctionModel() {}
+
+void ConvFunctionModel::clearData() {
+  m_fitType = FitType::None;
+  m_hasDeltaFunction = false;
+  m_backgroundType = BackgroundType::None;
+  m_model.clear();
+}
+
+void ConvFunctionModel::setModel() {
+  m_model.setModel(buildBackgroundFunctionString(), m_fitResolutions,
+                   buildPeaksFunctionString(), m_hasDeltaFunction, m_qValues,
+                   m_isQDependentFunction);
+  m_model.setGlobalParameters(makeGlobalList());
+}
+
+void ConvFunctionModel::setFunction(IFunction_sptr fun) {
+  clearData();
+  if (!fun)
+    return;
```

Comment:
Should this ever happen? If not, we should throw

Reply:
I think there are some cases where you might pass a nul pointer her if no function is set for whatever reason so would prefer to err on the safe side and leave this